### ISSUE TITLE
Save too slowly when there are too many pages

### ISSUE THREAD

**Issue by zyjdmmm** [xournalpp/xournalpp#5535]
### Is Your Feature Request Related to a Problem?

Save too slowly when there are too many pages,more than 200 pages often take more than 10 seconds, which is difficult to accept
![image](https://github.com/xournalpp/xournalpp/assets/75295100/918a20d6-053e-4526-bf58-e3924656ec39)

### Feature Description

I want to save only the changed pages, so it's fast

**Comment by vursc** [comment]
Since .xopp files are basically gzipped XML, implementing incremental saves is impossible without a radical change of the file format. However, the most time-consuming part of the save operation seems to be compression, which can be sped up by ~4x with a ~20% file size increase. Adding an option to save with faster compression would probably be possible.

**Comment by SJK9906** [comment]
i have this prole with my book of math in .pdf format while editing it

**Comment by eldipa** [comment]
> Since .xopp files are basically gzipped XML, implementing incremental saves is impossible without a radical change of the file format. 

Working on that! For the curious, we are tracking the progress in #937 

> However, the most time-consuming part of the save operation seems to be compression, which can be sped up by ~4x with a ~20% file size increase. Adding an option to save with faster compression would probably be possible.

It is true that the compression adds an overhead but depending of the content of the `.xopp` file it may not be the biggest problem. If the file has several Latex or images, those are getting copied over on save and that is a quite slow process.
I don't have the exact numbers but I remember a save operation with compression taking ~10 secs and without compression ~7 secs.

I hope #937 will change this for better.

**Comment by eldipa** [comment]
By the way, @zyjdmmm would you like to share your `.xopp` file ? (assuming that has no personal information).
I'm collecting `.xopp` files for evaluate the performance of the current file format and the next generation in #4862

**Comment by zyjdmmm** [comment]
@eldipa sure,here is!
[我的高数草稿本2.zip](https://github.com/xournalpp/xournalpp/files/14783586/2.zip)

**Comment by eldipa** [comment]
> @eldipa sure,here is! [我的高数草稿本2.zip](https://github.com/xournalpp/xournalpp/files/14783586/2.zip)

Thanks!

**Comment by eldipa** [comment]
> Only 10 seconds is good With multiple images (and my slow work pc it takes forever. But good to know it's being worked on. Thanks :D I could send an example file too if it helps

Yeah, I started to work on the new format exactly for that same reason!
